core/src/main/java/io/questdb/griffin/FunctionFactoryScanner.java

```diff
@@ -0,0 +1,308 @@
+        }
+
+        return factories;
+    }
+
+    @Nullable
+    private static FunctionFactory getClass(CharSequence className, @Nullable Log log) {
+        @SuppressWarnings("rawtypes") Class clazz;
+        try {
+            clazz = Class.forName(className.toString());
+        } catch (ClassNotFoundException e) {
+            return null;
+        }
+
+        if (FunctionFactory.class.isAssignableFrom(clazz) && !Modifier.isAbstract(clazz.getModifiers())) {
+            try {
+                // Instantiate the class and add to the list
+                @SuppressWarnings("unchecked") FunctionFactory instance = (FunctionFactory) clazz.getDeclaredConstructor().newInstance();
+                return instance;
+            } catch (NoSuchMethodException e) {
+                // It's ok, not a function factory but a wrapper
+            } catch (Exception e) {
+                if (log != null) {
+                    log.advisory().$("error loading function: ").$(className).$(", error: ").$(e).$();
+                } else {
+                    System.out.println("error loading function: " + className + ", error: " + e);
+                    e.printStackTrace(System.out);
+                }
+            }
+        }
+        return null;
+    }
+
+    // Helper method to get the path of the JAR or directory the current class is in
+
+    private static int getOrder(FunctionFactory f1, CharSequenceIntHashMap orderMap) {
+        int index = orderMap.keyIndex(f1.getClass().getName());
+        if (index < 0) {
+            return orderMap.valueAt(index);
+        }
+        // Unknown functions at the end, longest signature first
+        // If name of the signatures match then the functions with fewer arguments come last
+        // so that VARLARG functions are always last
+        return orderMap.size() + Math.abs(10000 - f1.getSignature().length());
+    }
+
+    private static CharSequenceIntHashMap loadFunctionOrderMap() {
+        var fileName = "function_list.txt";
+        var map = new CharSequenceIntHashMap();
+
+        try (InputStream inputStream = FunctionFactoryScanner.class.getClassLoader().getResourceAsStream(fileName)) {
+            if (inputStream != null) {
+                var lines = new String(inputStream.readAllBytes()).split("\n");
+
+                int order = 0;
+                for (var line : lines) {
+                    var trimmed = line.trim();
+                    if (!trimmed.isBlank() && !trimmed.startsWith("#")) {
+                        map.put(line.trim(), order++);
+                    }
+                }
+            }
+        } catch (IOException e) {
+            // return empty map
+        }
+        return map;
+    }
+
+    // Scan for class files in a directory, including subdirectories
+    private static void scanDirectory(List<FunctionFactory> functionFactories, String dirPath, String packageName, @Nullable Log log) {
+        String packagePath = packageName.replace('.', '/');
```

Comment:
Shouldn't we use `Files.SEPARATOR` here and on the next line?

Reply:
Java supports both '/' and '\' on all platforms as path separators. I'm adding a comment